### PR TITLE
MVP-274: Component docs

### PR DESCRIPTION
**Pull request by BradyMitch**
## 🎯 Summary

  
[MVP-274](https://citz-imb.atlassian.net/jira/browse/MVP-274)

Added docs for the components and hooks like how MUI does their component docs.

View @ https://dev.developer.gov.bc.ca/docs/default/component/citz-imb-sso-react-npm-package/

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](/CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

### CODE REVIEW

File: techdocs/docs/using-the-package/apis-&-components/usesso-actions/get-auth-header-value.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# getAuthorizationHeaderValue Action 
+
+The `getAuthorizationHeaderValue` action of the `useSSO` hook returns the value for the `Authorization` header so that requests can be made to protected routes on the backend.
+
+It is advised to use the [fetchProtectedRoute] function instead.
```
Comment (TaylorFries) [inline techdocs/docs/using-the-package/apis-&-components/usesso-actions/get-auth-header-value.md:5]:
```suggestion
!!! note "Note"
    It is advised to use the [fetchProtectedRoute] function instead.
```

File: techdocs/docs/using-the-package/apis-&-components/usesso.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+export const RoleMessage = () => {
+  const { hasRoles } = useSSO();
+
+  // User must have 'Admin' role.
+  if (hasRoles(['Admin'])) 
+    return <p>Admin's have access to moderation features.</p>;
+
+  // Users must have BOTH 'Member' and 'Commenter' roles.
+  // requireAllRoles option is true by default.
+  if (hasRoles(['Member', 'VIP']))
```
Comment (TaylorFries) [inline techdocs/docs/using-the-package/apis-&-components/usesso.md:65]:
```suggestion
  if (hasRoles(['Member', 'Commenter'])) 
```

File: techdocs/docs/using-the-package/apis-&-components/usesso-actions/has-roles.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+export const RoleMessage = () => {
+  const { hasRoles } = useSSO();
+
+  // User must have 'Admin' role.
+  if (hasRoles(['Admin'])) 
+    return <p>Admin's have access to moderation features.</p>;
+
+  // Users must have BOTH 'Member' and 'Commenter' roles.
+  // requireAllRoles option is true by default.
+  if (hasRoles(['Member', 'VIP']))
```
Comment (TaylorFries) [inline techdocs/docs/using-the-package/apis-&-components/usesso-actions/has-roles.md:21]:
```suggestion
  if (hasRoles(['Member', 'Commenter'])) 
```

File: techdocs/docs/using-the-package/apis-&-components/usesso.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+import { useSSO } from "@bcgov/citz-imb-sso-react";
+
+export const RoleMessage = () => {
+  const { hasRoles } = useSSO();
+
+  // User must have 'Admin' role.
+  if (hasRoles(['Admin'])) 
+    return <p>Admin's have access to moderation features.</p>;
+
+  // Users must have BOTH 'Member' and 'Commenter' roles.
```
Comment (BradyMitch) [inline techdocs/docs/using-the-package/apis-&-components/usesso.md:63]:
```suggestion
  // Users must have BOTH 'Member' and 'VIP' roles.
```

File: techdocs/docs/using-the-package/apis-&-components/usesso.md (hunk 2 above)
Comment (BradyMitch) [reply techdocs/docs/using-the-package/apis-&-components/usesso.md:65]: I meant to change it to VIP but I forgot the comment. Changed that

File: techdocs/docs/using-the-package/apis-&-components/usesso-actions/has-roles.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import { useSSO } from "@bcgov/citz-imb-sso-react";
+
+export const RoleMessage = () => {
+  const { hasRoles } = useSSO();
+
+  // User must have 'Admin' role.
+  if (hasRoles(['Admin'])) 
+    return <p>Admin's have access to moderation features.</p>;
+
+  // Users must have BOTH 'Member' and 'Commenter' roles.
```
Comment (BradyMitch) [inline techdocs/docs/using-the-package/apis-&-components/usesso-actions/has-roles.md:19]:
```suggestion
  // Users must have BOTH 'Member' and 'VIP' roles.
```

**Review by TaylorFries — APPROVED** [review]
Techdocs look great in dev environment. all links work !

**Pull request merged**